src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -208,9 +211,25 @@ stop_metadata_sync_to_node(PG_FUNCTION_ARGS)
 						errmsg("node (%s,%d) does not exist", nodeNameString, nodePort)));
 	}
 
+	if (clearMetadata && NodeIsPrimary(workerNode))
+	{
+		/*
+		 * If this is a secondary node we can't actually clear metadata from it,
+		 * we assume the primary node is cleared.
+		 */
+		DropMetadataSnapshotOnNode(workerNode);
+	}
+
 	MarkNodeHasMetadata(nodeNameString, nodePort, false);
 	MarkNodeMetadataSynced(nodeNameString, nodePort, false);
 
+	char *extensionOwner = CitusExtensionOwnerName();
```

Comment:
Because `DropMetadataSnapshotOnNode` might not be called. It's determined by an optional parameter. However, this code block should be executed even when `clearMetadata = false` and `DropMetadataSnapshotOnNode` is not called.

Reply:
I see `local_group_id` as part of the `metadata`, that's why I think we should move the code inside `DropMetadataSnapshotOnNode()`.

Don't you see that as part of metadata?

Reply:
So you're suggesting that we should update the local group id, only when we clear the metadata?